integrationtests/src/test/java/io/kroxylicious/proxy/multitenant/BaseMultiTenantIT.java

```diff
@@ -0,0 +1,206 @@
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.assertj.core.api.Condition;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.io.TempDir;
+
+import io.kroxylicious.net.IntegrationTestInetAddressResolverProvider;
+import io.kroxylicious.proxy.BaseIT;
+import io.kroxylicious.proxy.config.ClusterNetworkAddressConfigProviderDefinitionBuilder;
+import io.kroxylicious.proxy.config.ConfigurationBuilder;
+import io.kroxylicious.proxy.config.FilterDefinitionBuilder;
+import io.kroxylicious.proxy.config.VirtualClusterBuilder;
+import io.kroxylicious.proxy.service.HostPort;
+import io.kroxylicious.test.tester.KroxyliciousTester;
+import io.kroxylicious.testing.kafka.api.KafkaCluster;
+import io.kroxylicious.testing.kafka.common.KeytoolCertificateGenerator;
+import io.kroxylicious.testing.kafka.junit5ext.KafkaClusterExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+@ExtendWith(KafkaClusterExtension.class)
+public abstract class BaseMultiTenantIT extends BaseIT {
+
+    public static final String TENANT_1_CLUSTER = "foo";
+    static final HostPort TENANT_1_PROXY_ADDRESS = HostPort
+            .parse(IntegrationTestInetAddressResolverProvider.generateFullyQualifiedDomainName(TENANT_1_CLUSTER, 9192));
+    public static final String TENANT_2_CLUSTER = "bar";
+    static final HostPort TENANT_2_PROXY_ADDRESS = HostPort
+            .parse(IntegrationTestInetAddressResolverProvider.generateFullyQualifiedDomainName(TENANT_2_CLUSTER, 9292));
+
+    static final long FUTURE_TIMEOUT_SECONDS = 5L;
+    Map<String, Object> clientConfig;
+
+    TestInfo testInfo;
+    KeytoolCertificateGenerator certificateGenerator;
+    @TempDir
+    Path certsDirectory;
+
+    void setup(TestInfo testInfo) throws Exception {
```

Comment:
absence of the @BeforeEach annotation looks suprising

Reply:
I always thought the magic Junit4 invocation of needed the method to be called `setUp`. 
```suggestion
    @BeforeEach
    void configureTrust(TestInfo testInfo) throws Exception {
```

Reply:
This now has the correct annotation, and is named `beforeEach` in line with the other integration tests. The name `configureTrust` didn't quite seem right as that's not the only thing this method does (we also instantiate `testInfo` and populate `clientConfig`).